connector/datadogconnector/connector.go

```diff
@@ -112,6 +128,31 @@ func (c *traceToMetricConnector) Capabilities() consumer.Capabilities {
 }
 
 func (c *traceToMetricConnector) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
+	if len(c.resourceAttrs) > 0 {
+		for i := 0; i < traces.ResourceSpans().Len(); i++ {
+			rs := traces.ResourceSpans().At(i)
+			attrs := rs.Resource().Attributes()
+			containerID, ok := attrs.Get(semconv.AttributeContainerID)
+			if !ok {
+				continue
+			}
+			ddContainerTags := attributes.ContainerTagsFromResourceAttributes(attrs)
+			for attr := range c.resourceAttrs {
+				if val, ok := ddContainerTags[attr]; ok {
+					var cacheVal map[string]struct{}
+					if v, ok := c.containerTagCache.Get(containerID.AsString()); ok {
+						cacheVal = v.(map[string]struct{})
+						cacheVal[fmt.Sprintf("%s:%s", attr, val)] = struct{}{}
```

Comment:
nit: can check the existence of the container tag first `_, exists := cacheVal(tag)`